.github/workflows/publish.yml

```diff
@@ -56,10 +57,74 @@ jobs:
+          OPENAI_AZURE_ENDPOINT: ${{ secrets.OPENAI_AZURE_ENDPOINT }} 
+          # CURRENT_TAG: ${{ steps.check_pypi.outputs.version }}
+          # PREVIOUS_TAG: ${{ steps.check_pypi.outputs.previous_version }}
+        shell: python
+        run: |
+          import openai
+          import os
+          import requests
+          import json
+
+          OPENAI_AZURE_API_KEY = os.getenv('OPENAI_AZURE_API_KEY')
+          GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+          OPENAI_AZURE_ENDPOINT = os.getenv('OPENAI_AZURE_ENDPOINT')
+          OPENAI_AZURE_API_VERSION = "2024-02-01"
+          # we generate a release note when there is a tag increment. For testing, we manually compare two different tags
+          # latest_tag = os.getenv('CURRENT_TAG') 
+          # previous_tag = os.getenv('PREVIOUS_TAG')
+          latest_tag = "v8.2.0"
+          previous_tag = "v8.1.0"
+          
+          repo = f'ultralytics/release-test' # change to main repo later
+          headers = {"Authorization": f"token {GITHUB_TOKEN}", "Accept": "application/vnd.github.v3+json"}
+
+          compare_tags_url = f"https://api.github.com/repos/{repo}/compare/{previous_tag}...{latest_tag}"
+
+          response = requests.get(compare_tags_url, headers=headers).json()
+          diffs = []
+          files = response.get('files', [])
+          for file in files:
+              patch = file.get('patch', '')
+              if patch:
+                  diffs.append(patch)       
+          combined_diff = '\n'.join(diffs)
+          limit = 64000 * 3 # can be changed if needed
+          client = openai.AzureOpenAI(api_key=OPENAI_AZURE_API_KEY, api_version=OPENAI_AZURE_API_VERSION, azure_endpoint=OPENAI_AZURE_ENDPOINT)
+          messages=[
+              {"role": "system", "content": "You are an Ultralytics AI assistant skilled in software development and technical communication. Your task is to summarize GitHub PRs from Ultralytics in a way that is accurate, concise, and understandable to both expert developers and non-expert users. Focus on highlighting the key changes and their impact in simple, concise terms."},
+              {"role": "user", "content": f"Summarize the updates made on '{latest_tag}' tag, focusing on major changes, their purpose, and potential impact. Keep the summary clear and concise, suitable for a broad audience. Add emojis to enliven the summary. Reply directly with a summary along these example guidelines, though feel free to adjust as appropriate:\n\n"
+              f"### 🌟 Summary (single-line synopsis)\n"
+              f"### 📊 Key Changes (bullet points highlighting any major changes)\n"
+              f"### 🎯 Purpose & Impact (bullet points explaining any benefits and potential impact to users)\n"
+              f"\n\nHere's the PR diff:\n\n{combined_diff[:limit]}",
+              }
+            ]
+          response = client.chat.completions.create(model="gpt-4-turbo-2024-04-09", messages=messages).choices[0]
```

Comment:
Can you try with gpt-4-0125-preview @IvorZhu331 ?

Reply:
Azure API is now working with `gpt-4-0125-preview`.